packages/MonorepoBuilder/packages/Split/src/PackageToRepositorySplitter.php

```diff
@@ -89,9 +96,12 @@ public function splitDirectoriesToRepositories(array $splitConfig, string $rootD
         $this->reportFinishedProcesses();
     }
 
-    private function processActiveProcesses(): void
+    private function processActiveProcesses(?int $numberOfProcessesToWaitFor = null): void
     {
-        while (count($this->activeProcesses)) {
+        $numberOfActiveProcesses = count($this->activeProcesses);
+        $numberOfProcessesToWaitFor = $numberOfProcessesToWaitFor ?? $numberOfActiveProcesses;
+
+        while ($numberOfActiveProcesses - count($this->activeProcesses) < $numberOfProcessesToWaitFor) {
```

Comment:
I don't understand what is going there. Could you extract a method?

Reply:
Yeah, I can do that. :+1:

Reply:
I simplified the logic, so I think a dedicated method is no longer needed: 0723e5141